source/slang/slang-parameter-binding.cpp

```diff
@@ -333,6 +333,8 @@ struct ParameterBindingInfo
 {
     size_t              space = 0;
     size_t              index = 0;
+    bool                hasInputAttachmentIndex = false;
+    size_t              inputAttachmentIndex = 0;
```

Comment:
This looks a lot of waste in memory.
You may need to find another way to avoid adding them in ParameterBindingInfo.

It looks like ParameterBindingAndKindInfo inherits from ParameterBindingInfo with "kind" information.
You may need to explorer how it works and see if you can piggy-back on that pattern.

LayoutSize is trying to avoid adding a boolean by handling one special case, "infinite".
If you cannot avoid adding the member variables, I think that's what you can do at least.
You could make a new class like that to indicate a special case of not having the value.

Reply:
I will see if I can treat this type the same way we manage 'layout' uniforms. The cost of memory is 12bytes*uniforms_in_shader.

I will address this if I can in the next commit

Reply:
It is 16 byes not 12. There is a concept of "padding". In short, the address of a member variable must be aligned to its size. In your case bool will use 4 byes and there will be 4 byes padding before the next size_t.

Reply:
I think the best case is to use a magic number approach. The life times of these objects are either limited, or most will be culled out, meaning we only use 8bytes*Uniform for a portion of the compile.

the alternative solutions don't improve the problem:
1. we could make a struct which inherits. This would work, but then when adding the actual code it will be a bunch of if statements everywhere for handling 1 of the 23 cases we currently have.
2. we could do a Vulkan style extension system. This would work, but then we would have to add a RefPtr<> since Slang does not use dangling pointers which is already substantially larger than a 8 byte pointer .

Reply:
This is not 8 bytes. It is still 16 bytes because the size of the struct must be the largest alignment required by its members.
In your case you are using 4 byes and there is unused 4 bytes as padding